libgstc/pygstc/tcp.py

```diff
@@ -127,45 +142,37 @@ def _recvall(self, sock, timeout):
         timeout : int
             Timeout in seconds to wait for a response. 0: infinite
 
+        Raises
+        ------
+        socket.error
+            Error is triggered when Gstd IPC fails
+        BufferError
+            When the incoming buffer is too big.
+
         Returns
         -------
         buf : string
             Raw socket response
         """
-
         buf = b''
-        count = 0
+        newbuf = ''
         try:
+            sock.settimeout(timeout)
             while True:
-                if self._maxsize:
-                    if count >= self._maxsize:
-                        break
-
-                # if timeout is set, perform non-blocking read
+                if (self._maxsize and self._maxsize > len(newbuf)):
+                    raise BufferError
+                # Timeout to perform non-blocking read
+                ready = select.select([sock], [], [], timeout)
```

Comment:
This should no longer be necessary.

Reply:
It is necessary, otherwise, we obtain a TimeoutError on the last reading for long messages.